ADAL/src/urlprotocol/ADNTLMHandler.m

```diff
@@ -98,7 +98,7 @@ + (BOOL)handleChallenge:(NSURLAuthenticationChallenge *)challenge
                 _challengeCancelled = YES;
                 AD_LOG_INFO_F(@"NTLM challenge cancelled", nil, @"host: %@", challenge.protectionSpace.host);
                 [challenge.sender performDefaultHandlingForAuthenticationChallenge:challenge];
-                [protocol stopLoading];
+                [protocol connection:connection didFailWithError:[NSError errorWithDomain:NSCocoaErrorDomain code:NSUserCancelledError userInfo:nil]];
             }
```

Comment:
without this line, webview won't return after user click "cancel" in NTLM dialog.